### PR TITLE
Class ts engine

### PR DESCRIPTION
**Pull request by uslava77**
u know what this is

### CODE REVIEW

File: src/codegen/TypeScriptEngine.ts [hunk 1]
```diff
@@ -33,12 +43,30 @@ export default class TypeScriptEngine {
         )
     }
 
-    // TODO: create class method.
+    public createClass(classDecl: ClassDecl): ClassDeclaration {
+        // Add function declarations
+        let classMembers: ClassElement[] =
+            classDecl.functions.map((func: FuncDecl) => this.createMethod(func));
```
Comment (jyoo980) [inline src/codegen/TypeScriptEngine.ts:52]: Is `classDecl.functions` ever going to be `undefined` or `null`?

**Review by jyoo980 — APPROVED** [review]
@uslava77 @scveloso 

It's a beautiful thing. I also took the output of your tests and slipped them into `TestScript.ts`. Looks great!

image

File: src/codegen/TypeScriptEngine.ts (hunk 1 above)
Comment (scveloso) [reply src/codegen/TypeScriptEngine.ts:52]: Nope currently we make a empty array in parse regardless of whether there are functions or not in the program.

Comment (jyoo980) [reply src/codegen/TypeScriptEngine.ts:52]: Yeet!

**Review by scveloso — APPROVED** [review]

**Pull request merged**